pkg/operator/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go

```diff
@@ -209,6 +217,71 @@ func (n *clusterManagerController) sync(ctx context.Context, controllerContext f
 	}
 	managementClient := n.operatorKubeClient // We assume that operator is always running on the management cluster.
 
+	if config.MWReplicaSetEnabled {
+		// If the work driver is not kube, we need to sync the work driver config secret
+		if workDriver != operatorapiv1.WorkDriverTypeKube {
+			operatorNamespace, err := helpers.GetOperatorNamespace()
+			if err != nil {
+				return err
+			}
+			// check the secret containing work driver config explicitly because
+			// resourceapply.SyncSecret below won't return err if the source secret is not found
+			_, err = n.operatorKubeClient.CoreV1().Secrets(operatorNamespace).Get(ctx, helpers.WorkDriverConfig, metav1.GetOptions{})
+			if err != nil {
+				if errors.IsNotFound(err) {
+					klog.Errorf("Secret %s not found in namespace %s", helpers.WorkDriverConfig, operatorNamespace)
+					conditionChanged := meta.SetStatusCondition(&clusterManager.Status.Conditions, metav1.Condition{
+						Type:    workDriverConfigSecretSynced,
+						Status:  metav1.ConditionFalse,
+						Reason:  "WorkDriverConfigSecretNotFound",
+						Message: "Work driver config secret is not found",
+					})
+					if conditionChanged {
+						clusterManager.Status.ObservedGeneration = clusterManager.Generation
+						_, err := n.patcher.PatchStatus(ctx, clusterManager, clusterManager.Status, originalClusterManager.Status)
+						if err != nil {
+							return err
+						}
+					}
+				}
+				return err
+			} else {
+				_, _, err = resourceapply.SyncSecret(ctx, n.operatorKubeClient.CoreV1(), n.recorder,
+					operatorNamespace, helpers.WorkDriverConfig, clusterManagerNamespace, helpers.WorkDriverConfig,
+					[]metav1.OwnerReference{})
+				if err != nil {
+					return err
+				}
+				conditionChanged := meta.SetStatusCondition(&clusterManager.Status.Conditions, metav1.Condition{
+					Type:    workDriverConfigSecretSynced,
+					Status:  metav1.ConditionTrue,
+					Reason:  "WorkDriverConfigSecretSynced",
+					Message: "Work driver config secret is synced",
+				})
+				if conditionChanged {
+					clusterManager.Status.ObservedGeneration = clusterManager.Generation
+					_, err := n.patcher.PatchStatus(ctx, clusterManager, clusterManager.Status, originalClusterManager.Status)
+					if err != nil {
+						return err
+					}
+				}
+			}
+		} else {
```

Comment:
why we need to remove it? The else section here seems unnecessary to me. we can just return a condition with true when kube-driver is set.

Reply:
This is used for case: non-kube driver is changed to `kube` driver, the condition should be removed, make sense?

Reply:
I think we can always add the condition.

Reply:
done, removed this part.